frontends/PyCDE/test/test_bit_pat.py

```diff
@@ -0,0 +1,79 @@
+# RUN: %PYTHON% %s | FileCheck %s
+
+from pycde import Input, Output, generator, Module
+from pycde.types import Bits, SInt, UInt
+from pycde.testing import unittestmodule
+
+from pycde.bit_pat import BitPat, dict_lookup
+
+# CHECK-LABEL:  msft.module @TestBitPat {} (%inp1: ui36, %inp2: si25, %inp3: i32) -> (out1: i32, out2: i32, match: i1, no_match: i1)
+# CHECK-NEXT:     %c3_i32 = hw.constant 3 : i32
+# CHECK-NEXT:     %c28799_i32 = hw.constant 28799 : i32
+# CHECK-NEXT:     %c4099_i32 = hw.constant 4099 : i32
+# CHECK-NEXT:     %c28799_i32_0 = hw.constant 28799 : i32
+# CHECK-NEXT:     %c8195_i32 = hw.constant 8195 : i32
+# CHECK-NEXT:     %c28799_i32_1 = hw.constant 28799 : i32
+# CHECK-NEXT:     %c16387_i32 = hw.constant 16387 : i32
+# CHECK-NEXT:     %c28799_i32_2 = hw.constant 28799 : i32
+# CHECK-NEXT:     %c20483_i32 = hw.constant 20483 : i32
+# CHECK-NEXT:     %c28799_i32_3 = hw.constant 28799 : i32
+# CHECK-NEXT:     %c2989_i32 = hw.constant 2989 : i32
+# CHECK-NEXT:     %c64206_i32 = hw.constant 64206 : i32
+# CHECK-NEXT:     %c3054_i32 = hw.constant 3054 : i32
+# CHECK-NEXT:     %c48879_i32 = hw.constant 48879 : i32
+# CHECK-NEXT:     %c3245_i32 = hw.constant 3245 : i32
+# CHECK-NEXT:     %0 = hwarith.cast %inp1 : (ui36) -> i32
+# CHECK-NEXT:     %1 = comb.and bin %0, %c28799_i32 : i32
+# CHECK-NEXT:     %2 = comb.icmp bin eq %c28799_i32, %1 : i32
+# CHECK-NEXT:     %3 = hwarith.cast %inp2 : (si25) -> i32
+# CHECK-NEXT:     %4 = comb.and bin %3, %c28799_i32 : i32
+# CHECK-NEXT:     %5 = comb.icmp bin ne %c3_i32, %4 : i32
+# CHECK-NEXT:     %c19_i32 = hw.constant 19 : i32
+# CHECK-NEXT:     %c-1_i32 = hw.constant -1 : i32
+# CHECK-NEXT:     %c0_i32 = hw.constant 0 : i32
+# CHECK-NEXT:     %6 = comb.and bin %inp3, %c28799_i32 : i32
+# CHECK-NEXT:     %7 = comb.icmp bin eq %c28799_i32, %6 : i32
+# CHECK-NEXT:     %8 = comb.mux bin %7, %c2989_i32, %c0_i32 {sv.namehint = "mux_None_in0_in1"} : i32
```

Comment:
I generally edit out the namehints and replace them with regexes. Also, ideally you'd replace the `%#` with [substitution blocks](https://www.llvm.org/docs/CommandGuide/FileCheck.html#filecheck-string-substitution-blocks) though I've been lazy about that in my tests so I don't generally expect others to do this.